src/cpp/session/modules/quarto/SessionQuarto.cpp

```diff
@@ -208,25 +208,30 @@ void detectQuartoInstallation()
    s_quartoVersion = std::get<1>(userInstalled);
    bool prepend = std::get<2>(userInstalled);
 
+   std::string sysPath = core::system::getenv("PATH");
+   r::exec::RFunction sysSetenv("Sys.setenv");
+
    // always use user installed if it's there but subject to version check
    if (!s_userInstalledPath.isEmpty())
    {
-      if (std::get<1>(userInstalled) >= kQuartoRecommendedVersion)
+      if (s_quartoVersion >= kQuartoRecommendedVersion)
       {
-         s_quartoPath = std::get<0>(userInstalled);
-         s_quartoVersion = std::get<1>(userInstalled);
-         // prepend to path if RSTUDIO_QUARTO is defined
-         if (prepend)
+         s_quartoPath = s_userInstalledPath;
+         const std::string quartoPath = string_utils::utf8ToSystem(s_quartoPath.getParent().getAbsolutePath());
+
+         // prepend to path only if RSTUDIO_QUARTO is defined
+         core::system::addToPath(&sysPath, quartoPath, prepend);
```

Comment:
I don't think it matters with the end result but I think the prepend is a bit overloaded with its use. `RSTUDIO_QUARTO` being set means it has to be added to the path by setting `prepend` but it's also used as whether to add it to the beginning or the end of the path. It absolutely should be added to the path in this case though.

For the user installed Quarto that is not found through `RSTUDIO_QUARTO`, that is already on the path because that is how it was discovered. So this appears to add it to the path a second time in that case.

Reply:
On windows systems it was not getting added to the path at all; see my comment below, it seems in some cases it's detecting it as user installed quarto even when there is no separately installed quarto and the user installed quarto is finding the bundled quarto. This seems to be why on Windows, Sys.which("quarto") returns a path to the quarto binary, even though the binary is not on the path variable, and RSTUDIO_QUARTO is unset. I can't say I fully understand why this is the case, but getting quarto on the path for windows systems is critical for this release; it doesn't seem like user installed really refers to user installed in all cases here, but regardless of whether it's truly user installed or not, we need to make sure it's on the path.

On Mac systems thus seemed to be already on the path, so this is possibly adding it to the path twice; I can add some check here to skip if it's already on the path

Reply:
If this issue is only affecting Windows then maybe an `ifdef` would work. I guess if it's added twice to the path it's not going to affect anything since it would be prepended and take precedence. I'm just thinking of the case where the discovered Quarto is, for some reason, different than the one already on the path but I think it's covered here.